queue_processors/autoreduction_processor/post_process_admin.py

```diff
@@ -206,12 +214,62 @@ def specify_instrument_directories(self,
         # Specify directories where autoreduction output will go
         return temporary_directory + instrument_output_directory
 
-    def reduction_started(self):
-        """Log and update AMQ message to reduction started"""
-        logger.debug("Calling: %s\n%s",
-                     ACTIVEMQ_SETTINGS.reduction_started,
-                     self.message.serialize(limit_reduction_script=True))
-        self.client.send(ACTIVEMQ_SETTINGS.reduction_started, self.message)
+    @staticmethod
+    def write_and_readability_checks(directory_list, read_write):
+        """
+        Check a list of N directories for write/readability
+        :param directory_list: (list) directory list
+        :param read_write: (str) Read=R, Write=W
+        :return Error (Exception or ValueError) when something goes wrong
+        """
+        read_write = read_write.upper()
+        read_write_map = {"R": "read", "W": "write"}
+        try:
+            read_write_map[read_write]
+        except KeyError:
+            raise KeyError("Invalid read or write input: %s read_write argument must be either"
+                           " 'R' or 'W'" % read_write)
+
+        for location in directory_list:
+            if not os.access(location, getattr(sys.modules[os.__name__], f"{read_write}_OK")):
+                if not os.access(location, os.F_OK):
+                    problem = "does not exist"
+                else:
+                    problem = "no %s access", read_write_map[read_write]
+                raise OSError("Couldn't %s %s  -  %s" % (read_write_map[read_write],
+                                                         location,
+                                                         problem))
+            return True
+
+    def path_access_validate(self, should_be_writable, should_be_readable):
```

Comment:
I feel this name is misleading, I would expect only validation and not the creation of directories.

Reply:
This is not code i have changed. It is the result of merging Master into 625. I will consider changing this later. The purpose of the PR's focused on the refacter of `reduce()` given this is the core of autoreduction is not to yet change the processes taking place, but simply to break apart reduce into sub-methods to test first as this method is largely untested.